### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by midnighter95**

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -227,6 +227,8 @@ object cosim extends Module {
 
   class emulator(xLen: String) extends Module {
 
+    val ncores: Int = Runtime.getRuntime.availableProcessors()
```
Comment (sequencer) [inline build.sc:230]: In case of large size cores machine, limit it to 8.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.13.3/install
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: Coursier Cache
+        uses: coursier/cache-action@v5
+
+      - name: run riscv-tests
+        run: |
+          nix --experimental-features 'nix-command flakes' develop -c make riscv64tests && nix --experimental-features 'nix-command flakes' develop -c make riscv32tests
```
Comment (sequencer) [inline .github/workflows/ci.yml]: Where is 64 xlen.

**Review by sequencer — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ksco) [reply .github/workflows/ci.yml]: Maybe split this into two steps? The 32 bits one and the 64 bits one.

File: build.sc (hunk 1 above)
Comment (midnighter95) [reply build.sc:230]: Set it to 8 causes `%Error: /nix/store/n46a7nkil7clgvmljxw2cp7hwj40jczh-verilator-5.006/share/verilator/include/verilated.cpp:2499: VerilatedContext has 2 threads but model 'VTestBench' (instantiated as 'TOP') was Verilated with --threads 8.` in CI.
How about set it to 2?

Comment (ksco) [reply build.sc:230]: MIN(n, 8) should be fine.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (midnighter95) [reply .github/workflows/ci.yml]: Spliting it to into two steps sounds good.

File: build.sc (hunk 1 above)
Comment (sequencer) [reply build.sc:230]: limit it to 8

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Coursier Cache
+        uses: coursier/cache-action@v5
+
+      - name: run riscv-32tests
+        run: |
+          nix --experimental-features 'nix-command flakes' develop -c make riscv32tests
+
+      - name: run riscv-64tests
+        run: |
+          nix --experimental-features 'nix-command flakes' develop -c make riscv64tests
```
Comment (sequencer) [inline .github/workflows/ci.yml]: split them

**Review by sequencer — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (midnighter95) [reply .github/workflows/ci.yml]: You means to split them into two jobs?

File: build.sc (hunk 1 above)
Comment (midnighter95) [reply build.sc:230]: Now it's Min(nCores,8)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (midnighter95) [reply .github/workflows/ci.yml]: It has been done using matrix.

**Review by sequencer — APPROVED** [review]

**Pull request merged**